CRM/MembershipExtras/Helper/RecurringContributionHelper.php

```diff
@@ -0,0 +1,21 @@
+<?php
+
+class CRM_MembershipExtras_Helper_RecurringContributionHelper {
+
+  /**
+   * Checks if recurring contribution links to any membership.
+   */
+  public static function isRecurringContributionLinkToMembership($recurringContributionID) {
```

Comment:
I would rename the method to `isRecurringContributionLinkedToMembership`

also, to correctly check if the recurring contribution is linked to a membership, you will need to do it diffeenrrly given more than one recurring contribution can be linked to a membership (specially after autorenewals), basically you will to do something close to this:

```
SELECT X FROM civicrm_contribution c
INNER JOIN civicrm_membership_payment mp ON c.id  = mp.contribution_id 
WHERE c.contribution_recur_id = {Input_recur_id}
```

Reply:
@omarabuhussein  the method has been renamed, but I am not sure if I understand why we need to care about if there is any other recurring contribution links to a membership? As we just need to check if the given recurring contribution is linked to any membership?

Reply:
@erawat 

it is something like this:

1- You create a membership with payment plan, at that moment the membership `contribution_recur_id` will be set to the recurring contribution id that was created.

2- After a year, the autorenewal job renewed the membership and created new recurring contribution for the new term, but in Membershipextras we don't update the membership `contribution_recur_id`  field to point to the new recurring contribution, so at the moment it basically just points to the first recurring contribution that was created. 

3- Even if we update the autorenewal job to update the membership  `contribution_recur_id` to point to the new one, it means the link between the old recurring contribution and the membership will be lost given it now will be pointing to the new one. 

because of that, using the  membership `contribution_recur_id` field is not a reliable way to tell if the recurring contribution is linked to a membership, and the only way to do it is by doing a join on the civicrm_membership_payment and the civicrm_contribution tables.

Reply:
@erawat on a 2nd thought, I don't think we need this new ticket, even if we let Membershipextras update GoCardless recurring contributions cycle day, it is not like CiviCRM core makes any use of it.